### PR TITLE
Run /_execute in User context

### PR DESCRIPTION
**Pull request by skkosuri-amzn**
*Issue #, if available:*
Run /_execute in User context

*Description of changes:*
Run /_execute in User context

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt [hunk 1]
```diff
@@ -39,8 +41,14 @@ class TransportExecuteMonitorAction @Inject constructor(
 ) : HandledTransportAction<ExecuteMonitorRequest, ExecuteMonitorResponse> (
         ExecuteMonitorAction.NAME, transportService, actionFilters, ::ExecuteMonitorRequest) {
 
+    private var user: User? = null
+
     override fun doExecute(task: Task, execMonitorRequest: ExecuteMonitorRequest, actionListener: ActionListener<ExecuteMonitorResponse>) {
 
+        val userStr = client.threadPool().threadContext.getTransient<String>(ConfigConstants.OPENDISTRO_SECURITY_USER_AND_ROLES)
+        log.debug("User and roles string from thread context: $userStr")
```
Comment (qreshi) [inline alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt:49]: I don't remember if `log.debug()` prepends any context information but if it doesn't, could we make this message a little more specific to know that it's coming from the execute transport action?

Comment (skkosuri-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt:49]: Double checked, it looks like below: `[2020-12-01T00:10:12,201][DEBUG][c.a.o.a.t.TransportIndexMonitorAction] [38f9d35bb887.ant.amazon.com] User and roles string from thread context: admin|admin|own_index,all_access`

It has the transport class name.

**Review by lezzago — APPROVED** [review]

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt (hunk 1 above)
Comment (qreshi) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt:49]: Cool, thanks for checking

**Review by qreshi — APPROVED** [review]

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt [hunk 2]
```diff
@@ -87,7 +95,10 @@ class TransportExecuteMonitorAction @Inject constructor(
                     }
                 })
             } else {
-                val monitor = execMonitorRequest.monitor as Monitor
+                val monitor = when (user == null || user?.name.isNullOrEmpty()) {
+                    true -> execMonitorRequest.monitor as Monitor
+                    false -> (execMonitorRequest.monitor as Monitor).copy(user = user)
```
Comment (ylwu-amzn) [inline alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt:100]: If user has permission to run mointor, then they can run other user's monitor?

Comment (skkosuri-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt:100]: The flow is same, if User roles has permission to run monitor and read permission for the index being monitored, yes.

Comment (ylwu-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt:100]: If user enabled user backend role filter for alerting, then they should not see/run other user's monitor? Even user has read permission for the index being monitored and run monitor permission, user should not be able to see other user's monitor on Kibana, so they can't execute other user's monitor from Kibana. But if they know other user's monitor id, they can execute the monitor?
What problem will this PR address ? Per my understanding, this PR will fix such case: user in monitor has no run monitor permission, but the person who execute the monitor has run monitor permission, for such case, this PR will allow that person to run monitor successfully.

**Pull request merged**